subsys/net/ip/l2/ieee802154/ieee802154_security.c

```diff
@@ -108,14 +108,21 @@ bool ieee802154_decrypt_auth(struct ieee802154_security_ctx *sec_ctx,
 	nonce[12] = sec_ctx->level;
 
 	pkt.in_buf = decrypt_payload_len ? frame + auth_payload_len : NULL;
-	pkt.in_len = decrypt_payload_len;
-	pkt.out_buf = frame;
+	/* payload lengths without the length of the tag */
+	pkt.in_len = decrypt_payload_len -
+		     sec_ctx->dec.mode_params.ccm_info.tag_len;
+	/*
+	 * TC does not copy the header, so prevent the header from being
+	 * overwritten.
+	 */
```

Comment:
Please file bugs for the crypto subsystem: users of the API should never make any assumptions on how they're implemented.  This seems to be an issue for the tag for the CCM operation in mbedTLS below as well.

Reply:
This is wrong formulated (sorry, I tested TC first), mbedTLS behaves the same way. That one with the tag is not so nice and mbedTLS does not check it, is also noted in [samples/drivers/crypto](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/drivers/crypto/src/main.c#L321)

Reply:
You don't need that comment. That said: if crypto api does not describes how fields should be filled-in then it needs to be fixed there as well.

Also, you are probably breaking cc2520 crypto device compatibility, you'll have to fix it